ddtrace/contrib/internal/ray/patch.py

```diff
@@ -0,0 +1,439 @@
+from .utils import _inject_context_in_kwargs
+from .utils import _inject_dd_trace_ctx_kwarg
+from .utils import _inject_ray_span_tags_and_metrics
+from .utils import extract_signature
+from .utils import get_dd_job_name_from_entrypoint
+from .utils import get_dd_job_name_from_submission_id
+from .utils import set_maybe_big_tag
+
+
+config._add(
+    "ray",
+    dict(
+        _default_service=schematize_service_name("ray"),
+        resubmit_interval=_get_config("DD_TRACE_RAY_RESUBMIT_LONG_RUNNING_INTERVAL", default=120.0, modifier=float),
+        initial_submit_threshold=_get_config("_DD_TRACE_RAY_INITIAL_SUBMIT_THRESHOLD", default=10.0, modifier=float),
+    ),
+)
+
+
+def _supported_versions() -> Dict[str, str]:
+    return {"ray": ">=2.46.0"}
+
+
+def get_version() -> str:
+    return str(getattr(ray, "__version__", ""))
+
+
+def _wrap_remote_function_execution(function):
+    """Inject trace context parameter into function signature"""
+
+    @wraps(function)
+    def wrapped_function(*args, **kwargs):
+        return _wrap_task_execution(function, *args, **kwargs)
+
+    return wrapped_function
+
+
+def _wrap_task_execution(wrapped, *args, **kwargs):
+    """
+    Wraps the actual execution of a Ray task to trace its performance.
+    """
+    if DD_TRACE_CTX not in kwargs:
+        return wrapped(*args, **kwargs)
+
+    # Extract context from parent span
+    extracted_context = _TraceContext._extract(kwargs[DD_TRACE_CTX])
+    kwargs.pop(DD_TRACE_CTX)
+
+    with long_running_ray_span(
+        f"{wrapped.__module__}.{wrapped.__qualname__}",
+        service=os.environ.get(RAY_JOB_NAME),
+        span_type=SpanTypes.RAY,
+        child_of=extracted_context,
+        activate=True,
+    ) as task_execute_span:
+        try:
+            set_maybe_big_tag(task_execute_span, "ray.task.args", args)
+            set_maybe_big_tag(task_execute_span, "ray.task.kwargs", kwargs)
+
+            result = wrapped(*args, **kwargs)
+
+            task_execute_span.set_tag_str("ray.task.status", "success")
+            return result
+        except BaseException:
+            task_execute_span.set_tag_str("ray.task.status", "error")
+            raise
+
+
+def traced_submit_task(wrapped, instance, args, kwargs):
+    """Trace task submission, i.e the func.remote() call"""
+    if tracer.current_span() is None:
+        tracer.context_provider.activate(_extract_tracing_context_from_env())
+
+    # Inject dd_trace_ctx args in the function being executed by ray
+    with instance._inject_lock:
+        if instance._function_signature is None:
+            instance._function = _wrap_remote_function_execution(instance._function)
+            instance._function.__signature__ = _inject_dd_trace_ctx_kwarg(instance._function)
+            instance._function_signature = extract_signature(instance._function)
+
+    with tracer.trace(
+        f"{instance._function_name}.remote()", service=os.environ.get(RAY_JOB_NAME), span_type=SpanTypes.RAY
```

Comment:
Nit: We usually don't include brackets in operation names. Not sure if this is documented anywhere
```suggestion
        f"{instance._function_name}.remote", service=os.environ.get(RAY_JOB_NAME), span_type=SpanTypes.RAY
```

Reply:
addressed in [936e2f8](https://github.com/DataDog/dd-trace-py/pull/14511/commits/936e2f8879995831232f579c7b92f200c53d0e01)